pkg/registry/orasinterface.go

```diff
@@ -0,0 +1,39 @@
+package registry
+
+import (
+	"context"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/content"
+	orasregistry "oras.land/oras-go/v2/registry"
+	"oras.land/oras-go/v2/registry/remote"
+)
+
+// OrasInterface thin layer for oras.
+type OrasInterface interface {
+	Repository(ctx context.Context, reg *remote.Registry, name string) (orasregistry.Repository, error)
+	Resolve(ctx context.Context, srcStorage orasregistry.Repository, reference string) (ocispec.Descriptor, error)
+	CopyGraph(ctx context.Context, src content.ReadOnlyStorage, dst content.Storage, root ocispec.Descriptor, opts oras.CopyGraphOptions) error
+}
+
+// OrasImplementation thin wrapper on oras.
+type OrasImplementation struct{}
```

Comment:
curious why you decided to create this layer on top of oras.
I was thinking instead we could just define the interface with same same signature as the oras original methods
And then compose those original oras objects into one to satisfy that interface

like
```go
type OrasInterface interface {
    Repository(ctx context.Context, name string) (orasregistry.Repository, error)
    Resolve(ctx context.Context, reference string) (ocispec.Descriptor, error)
    CopyGraph(ctx context.Context, src content.ReadOnlyStorage, dst content.Storage, root ocispec.Descriptor, opts oras.CopyGraphOptions) error
}
```

Although it's still coupled to the oras module, since it uses its structs as input and output, it still can be implemented in different ways since it's not dependent on the "service" objects (`remote.Registry`, `rasregistry.Repository`, etc.).

Reply:
I only did this to get enough test coverage.

Reply:
Ah got you. Then if it doesn't require too much work, I would to change this to something similar to what I suggested. I believe it should allow us to reach the same amount of coverage.

Reply:
Yeh, compose isn't going to work, but I was going to rearrange the code a bit to get the coverage back up AND get rid of this OrasImplementation